modules/bot-improvement/src/views/full/index.tsx

```diff
@@ -0,0 +1,199 @@
+    const initializeState = async () => {
+      const qnaItems = await api.getQnaItems()
+      const goals = await api.getGoals()
+
+      const defaultQnaItemId = getDefaultQnaItemId(qnaItems)
+
+      const feedbackItems = (await api.getFeedbackItems()).map(i => {
+        i.correctedActionType = i.correctedActionType || 'qna'
+        i.correctedObjectId = i.correctedObjectId || defaultQnaItemId
+        i.status = i.status || 'pending'
+        return i
+      })
+
+      setState(state => ({ ...state, feedbackItemsLoading: false, feedbackItems, goals, qnaItems }))
+    }
+    initializeState().catch(e => {
+      throw e
+    })
+  }, [])
+
+  if (feedbackItems.length === 0) {
+    return (
+      <div style={{ width: '100%', height: '100%', display: 'flex', placeContent: 'center' }}>
+        <Callout title={'No feedback items'} style={{ width: '30%', margin: 'auto' }}>
+          Feedback items are created when chat users give negative feedback on bot messages
+        </Callout>
+      </div>
+    )
+  }
+
+  if (feedbackItemsLoading) {
+    return <Callout>Loading...</Callout>
+  }
+
+  const pendingFeedbackItems = feedbackItems.filter(i => i.status === 'pending')
+  const solvedFeedbackItems = feedbackItems.filter(i => i.status === 'solved')
+
+  if (!currentFeedbackItem) {
+    currentFeedbackItem = pendingFeedbackItems[0]
+  }
+
+  const updateFeedbackItem = async (item: FeedbackItem, changedProps) => {
+    const listClone = [...feedbackItems]
+    const itemClone = _.cloneDeep(item)
+    _.merge(itemClone, changedProps)
+
+    const idx = listClone.findIndex(e => e.eventId === item.eventId)
+    listClone[idx] = itemClone
+    setState(state => ({ ...state, feedbackItems: listClone }))
+
+    const { status, eventId, correctedActionType, correctedObjectId } = itemClone
+    await api.updateFeedbackItem({
+      status,
+      eventId,
+      correctedActionType,
+      correctedObjectId
+    })
+  }
+
+  const FeedbackItemPanel: FC<{ feedbackItems: FeedbackItem[] }> = props => {
+    const { feedbackItems } = props
+    return (
+      <div>
+        {feedbackItems.map(item => {
+          const handleCorrectedActionTypeChange = async (correctedActionType: string) => {
+            const defaultQnaItemId = getDefaultQnaItemId(qnaItems)
+            const defaultGoalId = getDefaultGoalId(goals)
+
+            await updateFeedbackItem(item, {
+              correctedActionType,
+              correctedObjectId: correctedActionType === 'qna' ? defaultQnaItemId : defaultGoalId
+            })
+          }
+
+          const handleCorrectedActionObjectIdChange = async (correctedObjectId: string) => {
+            await updateFeedbackItem(item, { correctedObjectId })
+          }
+
+          const markAsSolved = async () => {
+            await updateFeedbackItem(item, { status: 'solved' })
+          }
+
+          const markAsPending = async () => {
+            await updateFeedbackItem(item, { status: 'pending' })
+          }
```

Comment:
I think it would make more sense to move those 4 update methods directly in FeedbackItem, it would require less methods to pass to the child component